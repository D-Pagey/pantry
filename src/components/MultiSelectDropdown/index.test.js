/* eslint-disable no-shadow */
import React from 'react';
import userEvent from '@testing-library/user-event';
import { MultiSelectDropdown } from '.';

jest.mock('react-select/creatable', () => ({ options, value, onChange }) => {
    const handleChange = (event) => {
        const option = options.find((option) => option.value === event.currentTarget.value);
        onChange(option);
    };

    return (
        <select data-testid="select" value={value} onChange={handleChange}>
            {options.map(({ label, value }) => (
                <option key={value} value={value}>
                    {label}
                </option>
            ))}
        </select>
    );
});

const props = {
    options: [
        { label: 'Meat', value: 'meat' },
        { label: 'Fish', value: 'fish' }
    ],
    setValues: () => {},
    values: []
};

describe('MultiSelectDropdown component', () => {
    it('should render', () => {
        const { container } = render(<MultiSelectDropdown {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render a label', () => {
        const label = 'Which food categories is it?';
        const { getByText } = render(<MultiSelectDropdown {...props} label={label} />);
        getByText(label);
    });

    it('should render an error', () => {
        const error = 'Required';
        const { getByText } = render(<MultiSelectDropdown {...props} error={error} />);
        getByText(error);
    });

    it('should call setValues when changed', () => {
        const setValues = jest.fn();
        const option = { label: 'Fish', value: 'fish' };
        const { getByTestId } = render(<MultiSelectDropdown {...props} setValues={setValues} />);

        userEvent.selectOptions(getByTestId('select'), option.value);

        expect(setValues).toHaveBeenCalledWith(option);
    });
});
