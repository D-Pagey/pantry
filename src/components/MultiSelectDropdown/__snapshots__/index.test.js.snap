// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectDropdown component should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="c0"
>
  <select
    data-testid="select"
  >
    <option
      value="meat"
    >
      Meat
    </option>
    <option
      value="fish"
    >
      Fish
    </option>
  </select>
</div>
`;
